### PR TITLE
#152 - Fix for TrimmedPrefixString

### PR DESCRIPTION
**Pull request by rockfridrich**
Provides fixes for #152 with using of regular expression to match prefixes.

### CODE REVIEW

File: Web3Swift/StringScalar/TrimmedPrefixString.swift [hunk 1]
```diff
@@ -38,15 +40,10 @@ public final class TrimmedPrefixString: StringScalar {
...
-        if string.hasPrefix(prefix) {
-            return String(
-                string.dropFirst(
-                    Int(prefix.count)
-                )
-            )
-        } else {
-            return string
-        }
+        let pattern = "^(\(prefix)){1,}"
```
Comment (biboran) [inline Web3Swift/StringScalar/TrimmedPrefixString.swift]: Please don't use [redundant variables](https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html)

File: Web3Swift/StringScalar/TrimmedPrefixString.swift [hunk 2]
```diff
@@ -38,15 +40,10 @@ public final class TrimmedPrefixString: StringScalar {
...
+        guard let range = string.range(of: pattern, options: [.regularExpression]) else { return string }
```
Comment (biboran) [inline Web3Swift/StringScalar/TrimmedPrefixString.swift]: I think regex is unnecessary overkill in this case. You can simply do something like (pseudocode): 
```
let suffix = string.suffix(string.count % prefix.count)
while string.hasPrefix(prefix): string.dropFirst(prefix.count)
return string + suffix
```

Comment (biboran) [reply Web3Swift/StringScalar/TrimmedPrefixString.swift]: bonus points for immutable (`reduce`) version )

Comment (rockfridrich) [reply Web3Swift/StringScalar/TrimmedPrefixString.swift]: It looks less understandable than just one regex

File: Web3Swift/StringScalar/TrimmedPrefixString.swift [hunk 3]
```diff
@@ -38,15 +40,10 @@ public final class TrimmedPrefixString: StringScalar {
...
+        let trimmed = String(string[range.upperBound...])
+        return (!trimmed.isEmpty) ? trimmed : prefix
```
Comment (biboran) [inline Web3Swift/StringScalar/TrimmedPrefixString.swift]: Please replace ternary conditional operator with `if else`. Don't use `if not else`

File: Web3Swift/StringScalar/TrimmedPrefixString.swift [hunk 4]
```diff
@@ -38,15 +40,12 @@ public final class TrimmedPrefixString: StringScalar {
...
-            return String(
-                string.dropFirst(
-                    Int(prefix.count)
-                )
-            )
-        } else {
-            return string
+        guard let range = string.range(of: "^(\(prefix)){1,}", options: [.regularExpression]) else { return string }
+        let trimmed = String(string[range.upperBound...])
+        if (trimmed.isEmpty) {
```
Comment (biboran) [inline Web3Swift/StringScalar/TrimmedPrefixString.swift]: Please remove extraneous braces

Comment (rockfridrich) [reply Web3Swift/StringScalar/TrimmedPrefixString.swift]: Should it be a Swift Lint rule?

File: Web3Swift/StringScalar/TrimmedPrefixString.swift [hunk 5]
```diff
@@ -38,15 +40,12 @@ public final class TrimmedPrefixString: StringScalar {
...
         let prefix = try self.prefix.value()
-        if string.hasPrefix(prefix) {
-            return String(
-                string.dropFirst(
-                    Int(prefix.count)
-                )
-            )
-        } else {
-            return string
+        guard let range = string.range(of: "^(\(prefix)){1,}", options: [.regularExpression]) else { return string }
```
Comment (biboran) [inline Web3Swift/StringScalar/TrimmedPrefixString.swift]: Please put `of:`, `options:` and conditionals on separate lines. See `HexString.value()` for reference

File: Web3Swift/StringScalar/TrimmedPrefixString.swift (hunk 4 above)
Comment (biboran) [reply Web3Swift/StringScalar/TrimmedPrefixString.swift]: @rockfridrich yes

File: Example/Tests/StringScalar/TrimmedPrefixStringTests.swift [hunk 6]
```diff
@@ -17,29 +17,34 @@ final class TrimmedPrefixStringTests: XCTestCase {
     func testTrimsExistentPrefix() {
         Array<
             (
-                String,
-                String
+                String, //Full string
```
Comment (biboran) [inline Example/Tests/StringScalar/TrimmedPrefixStringTests.swift]: Please replace comments with tuple names

Comment (rockfridrich) [reply Example/Tests/StringScalar/TrimmedPrefixStringTests.swift]: Well it is pretty straightforward even without comment

**Pull request merged**